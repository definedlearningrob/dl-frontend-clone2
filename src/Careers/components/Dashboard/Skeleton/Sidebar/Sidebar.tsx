import { DCLogo } from '@dc/shared/DCLogo/DCLogo';

import SkeletonSidebarFooter from './Footer/Footer';
import SkeletonSidebarItem from './Item/Item';

type Props = {
  itemsCount: number;
};

function SkeletonSidebar({ itemsCount }: Props) {
  const renderItems = () =>
    Array.from({ length: itemsCount }, (_, index) => <SkeletonSidebarItem key={index} />);

  return (
    <div className='skeleton-sidebar'>
      <DCLogo className='skeleton-sidebar__logo-icon' />
      {renderItems()}
      <div className='skeleton-sidebar__divider' />
      <SkeletonSidebarFooter />
    </div>
  );
}

export default SkeletonSidebar;
