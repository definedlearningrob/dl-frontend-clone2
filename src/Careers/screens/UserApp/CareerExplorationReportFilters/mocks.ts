import { MockedResponse } from '@apollo/client/testing';
import {
  CareerExplorationReportFiltersDocument,
  CareerExplorationReportSummaryDocument,
  CareerExplorationReportVisitCountsDocument,
} from '@graphql/dc/users/hooks';

import { TCareerExplorationReportFiltersData } from '@dc/graphql/user/queries/careerExplorationReportFilters';

export const careerExplorationReportSummaryMock = {
  request: {
    query: CareerExplorationReportSummaryDocument,
    variables: {
      filter: { schoolYear: 2023, schoolClassUuids: undefined },
    },
  },
  result: {
    data: {
      reports: {
        pathwayReport: {
          summary: {
            engagementsCount: 10,
            clusterSCount: 2,
            pathwaysCount: 5,
            studentCount: 12,
            usersCount: 3,
          },
        },
      },
    },
  },
};

export const visitCountsSpy = jest.fn();

export const careerExplorationReportVisitCountsMock = {
  request: {
    query: CareerExplorationReportVisitCountsDocument,
    variables: {
      filter: { schoolYear: 2023, schoolClassUuids: undefined },
    },
  },
  result() {
    visitCountsSpy();

    return {
      data: {
        reports: {
          pathwayReport: [
            {
              visitsCount: 396,
              cluster: {
                name: 'Agriculture, Food & Natural Resources',
                id: '1',
                __typename: 'Cluster',
              },
              pathwayVisitCounts: [
                {
                  visitsCount: 54,
                  pathway: {
                    cluster: {
                      name: 'Agriculture, Food & Natural Resources',
                      id: '1',
                    },
                    id: '5',
                    name: 'Natural Resources Systems',
                  },
                },
                {
                  visitsCount: 55,
                  pathway: {
                    cluster: {
                      name: 'Agriculture, Food & Natural Resources',
                      id: '1',
                    },
                    id: '2',
                    name: 'Animal Systems',
                  },
                },
                {
                  visitsCount: 54,
                  pathway: {
                    cluster: {
                      name: 'Agriculture, Food & Natural Resources',
                      id: '1',
                    },
                    id: '6',
                    name: 'Plant Systems',
                  },
                },
              ],
            },
            {
              visitsCount: 10,
              cluster: {
                name: 'Transportation, Distribution & Logistics',
                id: '11',
              },
              pathwayVisitCounts: [
                {
                  visitsCount: 2,
                  pathway: {
                    cluster: {
                      name: 'Transportation, Distribution & Logistics',
                      id: '11',
                    },
                    id: '52',
                    name: 'Facility and Mobile Equipment Maintenance',
                  },
                },
                {
                  visitsCount: 3,
                  pathway: {
                    cluster: {
                      name: 'Transportation, Distribution & Logistics',
                      id: '11',
                    },
                    id: '49',
                    name: 'Transportation Operations',
                  },
                },
                {
                  visitsCount: 5,
                  pathway: {
                    cluster: {
                      name: 'Transportation, Distribution & Logistics',
                      id: '11',
                    },
                    id: '51',
                    name: 'Warehousing & Distribution Center Operations',
                  },
                },
              ],
            },
          ],
        },
      },
    };
  },
};

export const careerExplorationReportFiltersMock: MockedResponse<TCareerExplorationReportFiltersData> =
  {
    request: {
      query: CareerExplorationReportFiltersDocument,
      variables: {
        filters: {},
        entityFilter: { nameCont: '' },
        schoolClassFilter: { nameCont: '' },
        userFilter: { fullNameCont: '' },
      },
    },
    result: {
      data: {
        pathwayReportFilters: {
          entities: {
            nodesCount: 1,
            pagesCount: 1,
            nodes: [{ uuid: '1234-5678', name: 'City High School' }],
          },
          gradeLevels: ['10', '11', '12', 'Postsecondary'],
          users: {
            nodesCount: 1,
            pagesCount: 1,
            nodes: [{ uuid: '9876-5432', fullName: 'Anna Cityteach' }],
          },
          schoolClasses: {
            nodesCount: 0,
            pagesCount: 0,
            nodes: [],
          },
        },
      },
    },
  };
