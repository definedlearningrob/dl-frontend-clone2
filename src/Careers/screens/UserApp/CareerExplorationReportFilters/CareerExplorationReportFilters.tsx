import { useTranslation } from 'react-i18next';
import { useEffect, useState } from 'react';

import {
  CareerExplorationReportFiltersInputs,
  FINAL_STEP,
} from '@dc/components/CareerPathwayReport/CareerExplorationReportFiltersInputs';
import { useCareerExplorationReportFilters } from '@dc/components/CareerPathwayReport/useCareerExplorationReportFilters';

import careerPathwayReportImage from '@shared/assets/images/career_pathway.jpeg';
import { MainContent } from '@shared/components/MainContent/MainContent';
import { ReportFilterWrapper } from '@shared/components/ReportFilterWrapper/ReportFilterWrapper';
import { useNavigation } from '@shared/components/Sidebar/useNavigation';
import { REPORT_PATHS } from '@shared/resources/constants';

export const CareerExplorationReportFilters = () => {
  const { t } = useTranslation();
  const { setBackNavButton } = useNavigation();
  const { submitFilters } = useCareerExplorationReportFilters();
  const [isSubmitEnabled, setSubmitEnabled] = useState(false);

  useEffect(() => {
    setBackNavButton(true);

    return () => setBackNavButton(false);
  }, []);

  const handleStepChange = (activeStep: number) => {
    setSubmitEnabled(activeStep === FINAL_STEP);
  };

  return (
    <MainContent className='h-[theme(layout.containerHeight)]'>
      <ReportFilterWrapper
        filterInstruction={t('reports.filterInstructions')}
        isSubmitEnabled={isSubmitEnabled}
        redirectTo={`/reports/${REPORT_PATHS.CAREER_PATHWAY}/report`}
        reportImageSrc={careerPathwayReportImage}
        reportName={t('reports.reportName', {
          reportName: t('reports.reportTitle.careerPathway'),
        })}
        onFilterApply={submitFilters}>
        <CareerExplorationReportFiltersInputs onStepChange={handleStepChange} />
      </ReportFilterWrapper>
    </MainContent>
  );
};
