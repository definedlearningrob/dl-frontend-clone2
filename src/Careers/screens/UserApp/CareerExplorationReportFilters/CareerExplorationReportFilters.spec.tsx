import { fireEvent, screen, waitFor, within } from '@testing-library/dom';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';

import {
  CAREER_EXPLORATION_REPORT_FILTERS,
  TCareerExplorationReportFiltersData,
} from '@dc/graphql/user/queries/careerExplorationReportFilters';

import { renderWithRouter } from '@shared/utils/test';
import { ReportFiltersProvider } from '@shared/components/ReportFiltersProvider/ReportFiltersProvider';
import { ALL_OPTION } from '@shared/components/MultiSelect';
import { NavigationContextProvider } from '@shared/components/Sidebar/useNavigation';

import { CareerExplorationReportFilters } from './CareerExplorationReportFilters';

const initialFilters = {
  schoolYear: { label: '2023 / 2024', value: 2023 },
  entities: [ALL_OPTION],
  gradeLevels: [ALL_OPTION],
  users: [ALL_OPTION],
  schoolClasses: [ALL_OPTION],
  entityFilter: '',
  userFilter: '',
  schoolClassFilter: '',
};

const careerPathwayReportFiltersResult: TCareerExplorationReportFiltersData = {
  pathwayReportFilters: {
    entities: {
      nodesCount: 1,
      pagesCount: 1,
      nodes: [
        {
          uuid: '1234-5678',
          name: 'City High School',
        },
      ],
    },
    gradeLevels: ['10', '11', '12', 'Postsecondary'],
    users: {
      nodesCount: 5,
      pagesCount: 1,
      nodes: [
        {
          uuid: '0987-6543',
          fullName: 'Abigail Cteach',
        },
        {
          uuid: '9876-5432',
          fullName: 'Anna Cityteach',
        },
        {
          uuid: '8765-4321',
          fullName: 'Bertha Daniel',
        },
        {
          uuid: '7654-3210',
          fullName: 'Braeden Glover',
        },
        {
          uuid: '6543-2109',
          fullName: 'Brianna Greene',
        },
      ],
    },
    schoolClasses: {
      nodesCount: 0,
      pagesCount: 0,
      nodes: [],
    },
  },
};

const assessmentReportFiltersMock: MockedResponse<TCareerExplorationReportFiltersData> = {
  request: {
    query: CAREER_EXPLORATION_REPORT_FILTERS,
    variables: {
      filters: {},
      entityFilter: { nameCont: '' },
      schoolClassFilter: { nameCont: '' },
      userFilter: { fullNameCont: '' },
    },
  },
  result: {
    data: careerPathwayReportFiltersResult,
  },
};

const renderComponent = (customMocks: MockedResponse[] = []) =>
  renderWithRouter(
    <MockedProvider mocks={[assessmentReportFiltersMock, ...customMocks]}>
      <NavigationContextProvider>
        <ReportFiltersProvider initialFilters={initialFilters}>
          <CareerExplorationReportFilters />
        </ReportFiltersProvider>
      </NavigationContextProvider>
    </MockedProvider>
  );

describe('CareerPathwayReportFilters', () => {
  it('should be initialized with correct value', () => {
    const { container } = renderComponent();

    const schoolsSelect = screen.getByTestId('report-filters-select-entities');
    const gradeLevelsSelect = screen.getByTestId('report-filters-select-gradeLevels');
    const teachersSelect = screen.getByTestId('report-filters-select-users');
    const classesSelect = screen.getByTestId('report-filters-select-schoolClasses');

    expect(within(schoolsSelect).getByTestId('select-chip')).toHaveTextContent('All');
    expect(within(gradeLevelsSelect).getByTestId('select-chip')).toHaveTextContent('All');
    expect(within(teachersSelect).getByTestId('select-chip')).toHaveTextContent('All');
    expect(within(classesSelect).getByTestId('select-chip')).toHaveTextContent('All');
    expect(screen.getByRole('combobox', { name: 'School Year 2023 / 2024' })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: 'Generate report' })).toBeEnabled();

    expect(container).toMatchSnapshot();
  });

  it('should disable "Generate report" button when filter is not selected', async () => {
    const refetchCareerPathwayFiltersSpy = jest.fn();

    renderComponent([
      {
        request: {
          query: CAREER_EXPLORATION_REPORT_FILTERS,
          variables: {
            filters: { gradeLevels: [] },
            entityFilter: { nameCont: '' },
            schoolClassFilter: { nameCont: '' },
            userFilter: { fullNameCont: '' },
          },
        },
        result() {
          refetchCareerPathwayFiltersSpy();

          return { data: careerPathwayReportFiltersResult };
        },
      },
    ]);

    const gradeLevelsSelect = screen.getByTestId('report-filters-select-gradeLevels');
    const clearButton = within(gradeLevelsSelect).getByRole('button', { name: 'Remove All' });

    fireEvent.click(clearButton);

    expect(screen.getByRole('button', { name: 'Generate report' })).toBeDisabled();

    await waitFor(() => {
      expect(refetchCareerPathwayFiltersSpy).toHaveBeenCalledTimes(1);
    });
  });
});
