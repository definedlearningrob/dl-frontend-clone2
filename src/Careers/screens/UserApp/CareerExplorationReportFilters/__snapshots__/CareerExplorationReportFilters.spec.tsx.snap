// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CareerPathwayReportFilters should be initialized with correct value 1`] = `
<div>
  <div
    class="p-base xxxl:p-md h-[theme(layout.containerHeight)]"
    id="read"
  >
    <div
      class="card h-full"
    >
      <div
        class="flex gap-2lg h-full"
      >
        <div
          class="relative w-[320px] xxxl:w-[520px] self-stretch"
        >
          <img
            class="w-full h-full object-cover rounded-sm bg-gradient-neutral"
            src="career_pathway.jpeg"
          />
          <div
            class="bg-gradient-neutral absolute inset-0 rounded-sm"
          />
        </div>
        <div
          class="flex flex-col flex-1"
        >
          <h2
            class="text-base xxxl:text-2lg mb-xs xxxl:mb-sm"
          >
            Career Exploration Report
          </h2>
          <p
            class="text-xs xxxl:text-sm mb-sm xxxl:mb-base"
          >
            Fill in all the required fields below in particular order to generate the report
          </p>
          <div
            class="flex gap-sm xxxl:gap-base flex-1"
          >
            <div
              class="pt-[29px]"
            >
              <div
                class="p-xs rounded-sm font-medium text-sm xxxl:text-base xxxl:p-x bg-success-500 text-white"
              >
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0"
                    data-testid="icon"
                  >
                    <div
                      class="h-base w-base"
                    />
                  </div>
                </div>
              </div>
              <div
                class="w-xxxs rounded-sm flex-1 m-auto my-xs bg-success-500"
                style="height: 21px;"
              />
              <div
                class="p-xs rounded-sm font-medium text-sm xxxl:text-base xxxl:p-x bg-success-500 text-white"
              >
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0"
                    data-testid="icon"
                  >
                    <div
                      class="h-base w-base"
                    />
                  </div>
                </div>
              </div>
              <div
                class="w-xxxs rounded-sm flex-1 m-auto my-xs bg-success-500"
                style="height: 21px;"
              />
              <div
                class="p-xs rounded-sm font-medium text-sm xxxl:text-base xxxl:p-x bg-success-500 text-white"
              >
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0"
                    data-testid="icon"
                  >
                    <div
                      class="h-base w-base"
                    />
                  </div>
                </div>
              </div>
              <div
                class="w-xxxs rounded-sm flex-1 m-auto my-xs bg-success-500"
                style="height: 21px;"
              />
              <div
                class="p-xs rounded-sm font-medium text-sm xxxl:text-base xxxl:p-x bg-success-500 text-white"
              >
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0"
                    data-testid="icon"
                  >
                    <div
                      class="h-base w-base"
                    />
                  </div>
                </div>
              </div>
              <div
                class="w-xxxs rounded-sm flex-1 m-auto my-xs bg-success-500"
                style="height: 21px;"
              />
              <div
                class="p-xs rounded-sm font-medium text-sm xxxl:text-base xxxl:p-x bg-success-500 text-white"
              >
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0"
                    data-testid="icon"
                  >
                    <div
                      class="h-base w-base"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex-1 flex flex-col gap-xs xxxl:gap-sm"
            >
              <label
                class="flex flex-col leading-lg min-w-[120px] gap-xs"
                data-testid="report-filters-select-entities"
              >
                <div
                  class="text-font-primary leading-lg flex gap-xxxs text-xs !text-font-secondary opacity-50"
                >
                  Schools
                </div>
                <div
                  class="select--is-disabled css-3iigni-container"
                >
                  <span
                    class="css-1f43avz-a11yText-A11yText"
                    id="react-select-2-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                    role="log"
                  />
                  <div
                    aria-disabled="true"
                    class="!rounded-sm text-font-secondary !cursor-pointer outline-none !shadow-none !min-h-0 !border-neutral-300 hover:!border-neutral-400 opacity-50 !bg-neutral-200 !border-neutral-400 select__control select__control--is-disabled css-16xfy0z-control"
                  >
                    <div
                      class="flex gap-xxs items-center !flex-nowrap !text-xs !p-xs !pr-0 h-[38px] select__value-container select__value-container--is-multi select__value-container--has-value css-3w2yfm-ValueContainer"
                    >
                      <div
                        class="!bg-neutral-200 !rounded-xs p-xxs py-xxxs !m-0 select__multi-value select__multi-value--is-disabled css-1p3m7a8-multiValue"
                      >
                        <div
                          class="flex gap-xxs items-center min-w-0 bg-neutral-300"
                          data-testid="select-chip"
                        >
                          <div
                            class="!p-0 !text-font-primary min-w-0 !text-xs !leading-base select__multi-value__label css-wsp0cs-MultiValueGeneric"
                          >
                            All
                          </div>
                          <div
                            aria-label="Remove All"
                            class="!p-0 hover:!bg-danger-600 !rounded-xs overflow-hidden basis-sm shrink-0 select__multi-value__remove css-12a83d4-MultiValueRemove"
                            role="button"
                          >
                            <div
                              class="flex"
                            >
                              <div
                                class="flex align-center justify-center grow-0 shrink-0 text-font-primary hover:text-white"
                                data-testid="icon"
                              >
                                <div
                                  class="h-sm w-sm"
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="!p-0 !m-0 !leading-[22px] h-[22px] select__input-container css-1h01tm3-Input"
                        data-value=""
                      >
                        <input
                          aria-activedescendant=""
                          aria-autocomplete="list"
                          aria-expanded="false"
                          aria-haspopup="true"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          class="select__input"
                          disabled=""
                          id="select-input-entities"
                          role="combobox"
                          spellcheck="false"
                          style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="!p-0 !pr-xs select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <div
                        aria-hidden="true"
                        class="select__indicator select__loading-indicator css-pogzpp-loadingIndicator"
                      >
                        <span
                          class="css-1xtdfmb-LoadingDot"
                        />
                        <span
                          class="css-zoievk-LoadingDot"
                        />
                        <span
                          class="css-x748d8-LoadingDot"
                        />
                      </div>
                      <div
                        aria-hidden="true"
                        class="!p-0 select__indicator select__dropdown-indicator css-1xc3v61-indicatorContainer"
                      >
                        <div
                          class="flex"
                        >
                          <div
                            class="flex align-center justify-center grow-0 shrink-0 text-neutral-400 text-neutral-800"
                            data-testid="icon"
                          >
                            <div
                              class="h-base w-base"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </label>
              <label
                class="flex flex-col leading-lg min-w-[120px] gap-xs"
                data-testid="report-filters-select-gradeLevels"
              >
                <div
                  class="text-font-primary leading-lg flex gap-xxxs text-xs !text-font-secondary opacity-50"
                >
                  Grade Levels
                </div>
                <div
                  class="select--is-disabled css-3iigni-container"
                >
                  <span
                    class="css-1f43avz-a11yText-A11yText"
                    id="react-select-3-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                    role="log"
                  />
                  <div
                    aria-disabled="true"
                    class="!rounded-sm text-font-secondary !cursor-pointer outline-none !shadow-none !min-h-0 !border-neutral-300 hover:!border-neutral-400 opacity-50 !bg-neutral-200 !border-neutral-400 select__control select__control--is-disabled css-16xfy0z-control"
                  >
                    <div
                      class="flex gap-xxs items-center !flex-nowrap !text-xs !p-xs !pr-0 h-[38px] select__value-container select__value-container--is-multi select__value-container--has-value css-3w2yfm-ValueContainer"
                    >
                      <div
                        class="!bg-neutral-200 !rounded-xs p-xxs py-xxxs !m-0 select__multi-value select__multi-value--is-disabled css-1p3m7a8-multiValue"
                      >
                        <div
                          class="flex gap-xxs items-center min-w-0 bg-neutral-300"
                          data-testid="select-chip"
                        >
                          <div
                            class="!p-0 !text-font-primary min-w-0 !text-xs !leading-base select__multi-value__label css-wsp0cs-MultiValueGeneric"
                          >
                            All
                          </div>
                          <div
                            aria-label="Remove All"
                            class="!p-0 hover:!bg-danger-600 !rounded-xs overflow-hidden basis-sm shrink-0 select__multi-value__remove css-12a83d4-MultiValueRemove"
                            role="button"
                          >
                            <div
                              class="flex"
                            >
                              <div
                                class="flex align-center justify-center grow-0 shrink-0 text-font-primary hover:text-white"
                                data-testid="icon"
                              >
                                <div
                                  class="h-sm w-sm"
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <input
                        aria-activedescendant=""
                        aria-autocomplete="list"
                        aria-expanded="false"
                        aria-haspopup="true"
                        aria-readonly="true"
                        class="css-mohuvp-dummyInput-DummyInput"
                        disabled=""
                        id="select-input-gradeLevels"
                        inputmode="none"
                        role="combobox"
                        tabindex="0"
                        value=""
                      />
                    </div>
                    <div
                      class="!p-0 !pr-xs select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <div
                        aria-hidden="true"
                        class="select__indicator select__loading-indicator css-pogzpp-loadingIndicator"
                      >
                        <span
                          class="css-1xtdfmb-LoadingDot"
                        />
                        <span
                          class="css-zoievk-LoadingDot"
                        />
                        <span
                          class="css-x748d8-LoadingDot"
                        />
                      </div>
                      <div
                        aria-hidden="true"
                        class="!p-0 select__indicator select__dropdown-indicator css-1xc3v61-indicatorContainer"
                      >
                        <div
                          class="flex"
                        >
                          <div
                            class="flex align-center justify-center grow-0 shrink-0 text-neutral-400 text-neutral-800"
                            data-testid="icon"
                          >
                            <div
                              class="h-base w-base"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </label>
              <label
                class="flex flex-col leading-lg min-w-[120px] gap-xs"
                data-testid="report-filters-select-users"
              >
                <div
                  class="text-font-primary leading-lg flex gap-xxxs text-xs !text-font-secondary opacity-50"
                >
                  Teachers
                </div>
                <div
                  class="select--is-disabled css-3iigni-container"
                >
                  <span
                    class="css-1f43avz-a11yText-A11yText"
                    id="react-select-4-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                    role="log"
                  />
                  <div
                    aria-disabled="true"
                    class="!rounded-sm text-font-secondary !cursor-pointer outline-none !shadow-none !min-h-0 !border-neutral-300 hover:!border-neutral-400 opacity-50 !bg-neutral-200 !border-neutral-400 select__control select__control--is-disabled css-16xfy0z-control"
                  >
                    <div
                      class="flex gap-xxs items-center !flex-nowrap !text-xs !p-xs !pr-0 h-[38px] select__value-container select__value-container--is-multi select__value-container--has-value css-3w2yfm-ValueContainer"
                    >
                      <div
                        class="!bg-neutral-200 !rounded-xs p-xxs py-xxxs !m-0 select__multi-value select__multi-value--is-disabled css-1p3m7a8-multiValue"
                      >
                        <div
                          class="flex gap-xxs items-center min-w-0 bg-neutral-300"
                          data-testid="select-chip"
                        >
                          <div
                            class="!p-0 !text-font-primary min-w-0 !text-xs !leading-base select__multi-value__label css-wsp0cs-MultiValueGeneric"
                          >
                            All
                          </div>
                          <div
                            aria-label="Remove All"
                            class="!p-0 hover:!bg-danger-600 !rounded-xs overflow-hidden basis-sm shrink-0 select__multi-value__remove css-12a83d4-MultiValueRemove"
                            role="button"
                          >
                            <div
                              class="flex"
                            >
                              <div
                                class="flex align-center justify-center grow-0 shrink-0 text-font-primary hover:text-white"
                                data-testid="icon"
                              >
                                <div
                                  class="h-sm w-sm"
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="!p-0 !m-0 !leading-[22px] h-[22px] select__input-container css-1h01tm3-Input"
                        data-value=""
                      >
                        <input
                          aria-activedescendant=""
                          aria-autocomplete="list"
                          aria-expanded="false"
                          aria-haspopup="true"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          class="select__input"
                          disabled=""
                          id="select-input-users"
                          role="combobox"
                          spellcheck="false"
                          style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="!p-0 !pr-xs select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <div
                        aria-hidden="true"
                        class="select__indicator select__loading-indicator css-pogzpp-loadingIndicator"
                      >
                        <span
                          class="css-1xtdfmb-LoadingDot"
                        />
                        <span
                          class="css-zoievk-LoadingDot"
                        />
                        <span
                          class="css-x748d8-LoadingDot"
                        />
                      </div>
                      <div
                        aria-hidden="true"
                        class="!p-0 select__indicator select__dropdown-indicator css-1xc3v61-indicatorContainer"
                      >
                        <div
                          class="flex"
                        >
                          <div
                            class="flex align-center justify-center grow-0 shrink-0 text-neutral-400 text-neutral-800"
                            data-testid="icon"
                          >
                            <div
                              class="h-base w-base"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </label>
              <span
                data-state="closed"
              >
                <label
                  class="flex flex-col leading-lg min-w-[120px] gap-xs"
                  data-testid="report-filters-select-schoolClasses"
                >
                  <div
                    class="text-font-primary leading-lg flex gap-xxxs text-xs !text-font-secondary opacity-50"
                  >
                    Classes
                  </div>
                  <div
                    class="select--is-disabled css-3iigni-container"
                  >
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-5-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                      role="log"
                    />
                    <div
                      aria-disabled="true"
                      class="!rounded-sm text-font-secondary !cursor-pointer outline-none !shadow-none !min-h-0 !border-neutral-300 hover:!border-neutral-400 opacity-50 !bg-neutral-200 !border-neutral-400 select__control select__control--is-disabled css-16xfy0z-control"
                    >
                      <div
                        class="flex gap-xxs items-center !flex-nowrap !text-xs !p-xs !pr-0 h-[38px] select__value-container select__value-container--is-multi select__value-container--has-value css-3w2yfm-ValueContainer"
                      >
                        <div
                          class="!bg-neutral-200 !rounded-xs p-xxs py-xxxs !m-0 select__multi-value select__multi-value--is-disabled css-1p3m7a8-multiValue"
                        >
                          <div
                            class="flex gap-xxs items-center min-w-0 bg-neutral-300"
                            data-testid="select-chip"
                          >
                            <div
                              class="!p-0 !text-font-primary min-w-0 !text-xs !leading-base select__multi-value__label css-wsp0cs-MultiValueGeneric"
                            >
                              All
                            </div>
                            <div
                              aria-label="Remove All"
                              class="!p-0 hover:!bg-danger-600 !rounded-xs overflow-hidden basis-sm shrink-0 select__multi-value__remove css-12a83d4-MultiValueRemove"
                              role="button"
                            >
                              <div
                                class="flex"
                              >
                                <div
                                  class="flex align-center justify-center grow-0 shrink-0 text-font-primary hover:text-white"
                                  data-testid="icon"
                                >
                                  <div
                                    class="h-sm w-sm"
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                        <div
                          class="!p-0 !m-0 !leading-[22px] h-[22px] select__input-container css-1h01tm3-Input"
                          data-value=""
                        >
                          <input
                            aria-activedescendant=""
                            aria-autocomplete="list"
                            aria-expanded="false"
                            aria-haspopup="true"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            class="select__input"
                            disabled=""
                            id="select-input-schoolClasses"
                            role="combobox"
                            spellcheck="false"
                            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class="!p-0 !pr-xs select__indicators css-1hb7zxy-IndicatorsContainer"
                      >
                        <div
                          aria-hidden="true"
                          class="select__indicator select__loading-indicator css-pogzpp-loadingIndicator"
                        >
                          <span
                            class="css-1xtdfmb-LoadingDot"
                          />
                          <span
                            class="css-zoievk-LoadingDot"
                          />
                          <span
                            class="css-x748d8-LoadingDot"
                          />
                        </div>
                        <div
                          aria-hidden="true"
                          class="!p-0 select__indicator select__dropdown-indicator css-1xc3v61-indicatorContainer"
                        >
                          <div
                            class="flex"
                          >
                            <div
                              class="flex align-center justify-center grow-0 shrink-0 text-neutral-400 text-neutral-800"
                              data-testid="icon"
                            >
                              <div
                                class="h-base w-base"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </label>
              </span>
              <label
                class="flex flex-col leading-lg min-w-[120px] gap-xs"
              >
                <div
                  class="text-font-primary leading-lg flex gap-xxxs text-xs"
                >
                  School Year
                </div>
                <div
                  class=" css-b62m3t-container"
                >
                  <span
                    class="css-1f43avz-a11yText-A11yText"
                    id="react-select-6-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                    role="log"
                  />
                  <div
                    class="!rounded-sm text-font-secondary !cursor-pointer outline-none !shadow-none !min-h-0 !border-neutral-300 hover:!border-neutral-400 !bg-white select__control css-13cymwt-control"
                  >
                    <div
                      class="flex gap-xxs items-center !flex-nowrap !text-xs !p-xs !pr-0 h-[38px] select__value-container select__value-container--has-value css-1fdsijx-ValueContainer"
                    >
                      <div
                        class="!m-0 !leading-[22px] h-[22px] select__single-value css-1dimb5e-singleValue"
                      >
                        2023 / 2024
                      </div>
                      <input
                        aria-activedescendant=""
                        aria-autocomplete="list"
                        aria-expanded="false"
                        aria-haspopup="true"
                        aria-readonly="true"
                        class="css-mohuvp-dummyInput-DummyInput"
                        id="select-input-schoolYear"
                        inputmode="none"
                        role="combobox"
                        tabindex="0"
                        value=""
                      />
                    </div>
                    <div
                      class="!p-0 !pr-xs select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <div
                        aria-hidden="true"
                        class="!p-0 select__indicator select__dropdown-indicator css-1xc3v61-indicatorContainer"
                      >
                        <div
                          class="flex"
                        >
                          <div
                            class="flex align-center justify-center grow-0 shrink-0 text-neutral-400 text-neutral-800"
                            data-testid="icon"
                          >
                            <div
                              class="h-base w-base"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <input
                    name="schoolYear"
                    type="hidden"
                    value="2023"
                  />
                </div>
              </label>
            </div>
          </div>
          <span
            class="block w-fit"
            data-state="closed"
          >
            <a
              href="/reports/career-exploration/report"
            >
              <button
                class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] self-start -primary bg-primary-500 hover:bg-primary-600 outline-primary-500 border-primary-500 py-xs px-base leading-[22px] border text-white"
                data-testid="button"
                type="button"
              >
                <div
                  class="flex"
                >
                  <div
                    class="flex align-center justify-center grow-0 shrink-0"
                    data-testid="icon"
                  >
                    <div
                      class="h-sm w-sm"
                    />
                  </div>
                </div>
                <span
                  class="items-center flex justify-center gap-xs"
                >
                  Generate report
                </span>
              </button>
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
